src/vulkan/vk-resource-heap.cpp

```
@@ -0,0 +1,114 @@
+Result ResourceHeapImpl::init()
+{
+    DeviceImpl* device = getDevice<DeviceImpl>();
+    const VulkanApi& api = device->m_api;
+
+    uint32_t memoryTypeBits = ~0u;
+    if (m_desc.kind == ResourceHeapKind::Buffers || m_desc.kind == ResourceHeapKind::All)
+    {
+        VkBufferUsageFlags usage =
+            VK_BUFFER_USAGE_STORAGE_BUFFER_BIT | VK_BUFFER_USAGE_TRANSFER_SRC_BIT | VK_BUFFER_USAGE_TRANSFER_DST_BIT;
+        if (api.m_extendedFeatures.vulkan12Features.bufferDeviceAddress)
+            usage |= VK_BUFFER_USAGE_SHADER_DEVICE_ADDRESS_BIT;
```

Cppcheck (2.21.0)

[warning] If memory allocation fails, then there is a possible null pointer dereference

(nullPointerOutOfMemory)